client/verta/verta/runtime.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+"""Classes and functions to enable context logging within predictions made against
+models on the Verta Platform.
+
+.. versionadded:: 0.22.0
+
+"""
+
+import json
+import threading
+from typing import Any, Dict, Optional
+
+
+_THREAD = threading.local()
+
+def _init_thread_logs() -> None:
+    """
+    Initialize our thread-local variable for storing logging context.
+    """
+    if not hasattr(_THREAD, 'logs'):
+        _THREAD.logs = dict()
+
+
+def _get_thread_logs() -> Dict[str, Any]:
+    """
+    Return the current thread-local logs or initialize a new dict.
+    """
+    _init_thread_logs()
+    return _THREAD.logs
+
+
+def _set_thread_logs(logs: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Sets the thread-local logs, overwriting any existing values.
+    """
+    _THREAD.logs = logs
+    return _THREAD.logs
+
+
+def _init_validate_flag() -> None:
+    """
+    Initialize our thread-local variable for the validation flag.
+    """
+    if not hasattr(_THREAD, 'validate'):
+        _THREAD.validate = False
+
+
+def _get_validate_flag() -> bool:
+    """
+    Return the current thread-local variable for validate or initialize a
+    new boolean.
+    """
+    _init_validate_flag()
+    return _THREAD.validate
+
+
+def _set_validate_flag(flag: bool) -> bool:
+    """
+    Sets the thread-local validate flag boolean value.
+    """
+    _THREAD.validate = flag
+    return _THREAD.validate
+
+
+def _validate_json(value: Any) -> str:
+    """
+    Attempt to serialize the provided value to JSON.
+    Raise an error on failure.
+    """
+    try:
+        return json.dumps(value)
+    except json.JSONDecodeError as json_err:
+        print("provided data cannot be serialized into JSON for logging.")
+        raise json_err
+    except TypeError as type_err:
+        print("provided data type cannot be converted to JSON for logging.")
+        raise type_err
+
+
+def log(key: str, value: Any) -> None:
+    """
+    Updates current logging context dict with provided key and value.
+    For use within a model's :meth:`~verta.registry.VertaModelBase.predict`
+    method to collect logging context.
+
+    Parameters
+    ----------
+    key : str
+        String value to use as data label (key) in logging context.
+    value : Any
+        Any JSON serializable value you wish to include in the context logs.
```

Comment:
Is this too paranoid of me? 😆 

```suggestion
        Any `JSON serializable <https://docs.python.org/3/library/json.html#json.JSONEncoder>`__
        value you wish to include in the context logs.
```

Reply:
nope, I like it.  Done